internal/services/oracle/db_nodes_data_source.go

```diff
@@ -0,0 +1,261 @@
+// Copyright © 2024, Oracle and/or its affiliates. All rights reserved
+
+package oracle
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/oracledatabase/2024-06-01/dbnodes"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/oracle/validate"
+
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+type DBNodesDataSource struct{}
+
+type DBNodesDataModel struct {
+	CloudVmClusterName string            `tfschema:"cloud_vm_cluster_name"`
+	DBNodes            []DBNodeDataModel `tfschema:"db_nodes"`
+	ResourceGroupName  string            `tfschema:"resource_group_name"`
+}
+
+type DBNodeDataModel struct {
+	AdditionalDetails          string `tfschema:"additional_details"`
+	BackupIPId                 string `tfschema:"backup_ip_id"`
+	BackupVnic2Id              string `tfschema:"backup_vnic_2_id"`
+	BackupVnicId               string `tfschema:"backup_vnic_id"`
+	CpuCoreCount               int64  `tfschema:"cpu_core_count"`
+	DbNodeStorageSizeInGbs     int64  `tfschema:"db_node_storage_size_in_gbs"`
+	DbServerId                 string `tfschema:"db_server_id"`
+	DbSystemId                 string `tfschema:"db_system_id"`
+	FaultDomain                string `tfschema:"fault_domain"`
+	HostIPId                   string `tfschema:"host_ip_id"`
+	Hostname                   string `tfschema:"hostname"`
+	LifecycleDetails           string `tfschema:"lifecycle_details"`
+	LifecycleState             string `tfschema:"lifecycle_state"`
+	MaintenanceType            string `tfschema:"maintenance_type"`
+	MemorySizeInGbs            int64  `tfschema:"memory_size_in_gbs"`
+	Ocid                       string `tfschema:"ocid"`
+	SoftwareStorageSizeInGb    int64  `tfschema:"software_storage_size_in_gb"`
+	TimeCreated                string `tfschema:"time_created"`
+	TimeMaintenanceWindowEnd   string `tfschema:"time_maintenance_window_end"`
+	TimeMaintenanceWindowStart string `tfschema:"time_maintenance_window_start"`
+	Vnic2Id                    string `tfschema:"vnic_2_id"`
+	VnicId                     string `tfschema:"vnic_id"`
+}
+
+func (d DBNodesDataSource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"resource_group_name": commonschema.ResourceGroupNameForDataSource(),
+
+		"cloud_vm_cluster_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validate.CloudVMClusterName,
+		},
```

Comment:
Where possible we should use the id of parent resources as arguments, this means that these two properties can be replaced by one because we can infer the resource group name from the ID
```suggestion
		"cloud_vm_cluster_id": {
			Type:         pluginsdk.TypeString,
			Required:     true,
			ValidateFunc: cloudvmclusters.ValidateCloudVMClusterID,
		},
```

Reply:
Updated.